lib/Controller/DashboardController.php

```diff
@@ -0,0 +1,71 @@
+<?php
+
+declare(strict_types=1);
+
+/**
+ * SPDX-FileCopyrightText: 2025 Nextcloud GmbH and Nextcloud contributors
+ * SPDX-License-Identifier: AGPL-3.0-or-later
+ */
+
+namespace OCA\Talk\Controller;
+
+use OCA\Talk\Exceptions\ParticipantNotFoundException;
+use OCA\Talk\ResponseDefinitions;
+use OCA\Talk\Service\DashboardService;
+use OCA\Talk\Service\ParticipantService;
+use OCA\Talk\Service\RoomFormatter;
+use OCP\AppFramework\Http;
+use OCP\AppFramework\Http\Attribute\BruteForceProtection;
+use OCP\AppFramework\Http\Attribute\PublicPage;
+use OCP\AppFramework\Http\DataResponse;
+use OCP\IRequest;
+use OCP\IUserSession;
+use Psr\Log\LoggerInterface;
+
+/**
+ * @psalm-import-type TalkRoom from ResponseDefinitions
+ */
+class DashboardController extends AEnvironmentAwareOCSController {
+	public function __construct(
+		string $appName,
+		IRequest $request,
+		protected IUserSession $userSession,
+		protected LoggerInterface $logger,
+		protected DashboardService $service,
+		protected ParticipantService $participantService,
+		protected RoomFormatter $formatter,
+	) {
+		parent::__construct($appName, $request);
+	}
+
+	/**
+	 * Get rooms that have events in the next 7 days
+	 * sorted by their start timestamp ascending
+	 *
+	 * Required capability: `dashboard-event-rooms`
+	 *
+	 * @return DataResponse<Http::STATUS_OK, list<?TalkRoom>, array{}>
```

Comment:
It's always a list of that, even when it's empty:
```suggestion
	 * @return DataResponse<Http::STATUS_OK, list<TalkRoom>, array{}>
```